lms/djangoapps/courseware/module_render.py

```diff
@@ -377,6 +458,13 @@ def handle_grade_event(block, event_type, event):
 
         dog_stats_api.increment("lms.courseware.question_answered", tags=tags)
 
+        # Fulfill any milestones for this content module now that all the data has been saved
+        _fulfill_content_milestones(
```

Comment:
One comment here -- I think it's important to discern between fulfilling of milestones and the entrance exam case which leads to the fulfilling of milestones.  I think the outer check should be to see if the milestones app is available, and if so we step into the method and begin working through the various fulfillment scenarios.  I'll write it up in the code so you can see what I mean.